### DISCUSSION TITLE
Working with uint16 images

### DISCUSSION THREAD

**Discussion by lilianabrandao** [albumentations-team/albumentations#1423] [Q&A]
I set a transformation pipeline to augment uint16 images. The augmented images [along with the original dataset] were then used  to train a deep learning model.

No issue arose during the transform operations or visual debugging of the augmented images. However, the deep learning model results were not as expected [very similiar to those achieved without data augmentation].

Later on, I realized that Albumentations is not prepared to handle uint16 images.

Could this explain the lack of improvement for the training with data augmentation? 
What are the "consequences"/impact of not using the correct data type in Albumentations pipelines?

**Comment by ternaus** [comment]
This is a good question.

Right now we do not strictly check that all applied transforms work properly with images that are not `unit8` and `grayscale` / `3-channel`. It is the most common use case, but not the only one as there are: `unit16`, `float32` and multichannel images

It leads to unexpected behavior when people try to apply some non standard inputs.

But we are planning to address this issue in the nearest releases.
